src/features/defi/providers/idle/components/IdleManager/Claim/components/Confirm.tsx

```diff
@@ -0,0 +1,211 @@
+import { Alert, AlertIcon, Box, Stack } from '@chakra-ui/react'
+import { ASSET_REFERENCE, toAssetId } from '@shapeshiftoss/caip'
+import { supportsETH } from '@shapeshiftoss/hdwallet-core'
+import { Confirm as ReusableConfirm } from 'features/defi/components/Confirm/Confirm'
+import { Summary } from 'features/defi/components/Summary'
+import {
+  DefiAction,
+  DefiParams,
+  DefiQueryParams,
+  DefiStep,
+} from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { useIdle } from 'features/defi/contexts/IdleProvider/IdleProvider'
+import qs from 'qs'
+import { useContext, useEffect, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Amount } from 'components/Amount/Amount'
+import { StepComponentProps } from 'components/DeFi/components/Steps'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectPortfolioCryptoHumanBalanceByAssetId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { IdleClaimActionType } from '../ClaimCommon'
+import { ClaimContext } from '../ClaimContext'
+import { ClaimableAsset } from './ClaimableAsset'
+
+export const Confirm = ({ onNext }: StepComponentProps) => {
+  const translate = useTranslate()
+  const { state, dispatch } = useContext(ClaimContext)
+  const { idle: idleInvestor } = useIdle()
+  const opportunity = state?.opportunity
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress: vaultAddress, assetReference } = query
+
+  const assetNamespace = 'erc20'
+
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference: vaultAddress,
+  })
+
+  const feeAssetId = toAssetId({
+    chainId,
+    assetNamespace: 'slip44',
+    assetReference: ASSET_REFERENCE.Ethereum,
+  })
+  const feeAsset = useAppSelector(state => selectAssetById(state, feeAssetId))
+  const feeMarketData = useAppSelector(state => selectMarketDataById(state, feeAssetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+
+  const feeAssetBalance = useAppSelector(state =>
+    selectPortfolioCryptoHumanBalanceByAssetId(state, { assetId: feeAsset?.assetId ?? '' }),
+  )
+
+  useEffect(() => {
+    ;(async () => {
+      if (!dispatch || !state.userAddress || !idleInvestor) {
+        return
+      }
```

Comment:
Might be better for this to live two lines up, outside of the IIAFE to not push an additional promise we could avoid to the event loop